app/src/main/java/com/github/polypoly/app/models/commons/LoadingModel.kt

```diff
@@ -24,5 +27,25 @@ abstract class LoadingModel: ViewModel() {
      */
     protected fun setLoading(loading: Boolean) {
         isLoading.postValue(loading)
+        if (isLoading.value == true && !loading) {
+            for (future in waitingForSyncPromise) {
+                future.complete(true)
+            }
+            waitingForSyncPromise.clear()
+        }
+    }
+
+    /**
+     * Registers a callback that will be called next time a data is synchronized with the storage
+     * This function is primarily but not exclusively aimed for used in unit tests
+     * @return a promise that completes when the data is synced again with the storage
+     */
+    fun waitForSync(): CompletableFuture<Boolean> {
+        return if (isLoading.value == false) {
```

Comment:
this is odd to read, you could remove `== false` and invert the if/else bodies...